token/services/identity/msp/x509/deserializer.go

```diff
@@ -80,3 +84,51 @@ func (a *AuditInfoDeserializer) DeserializeAuditInfo(raw []byte) (*AuditInfo, er
 	}
 	return ai, nil
 }
+
+func GetEnrollmentID(id []byte) (string, error) {
+	si := &msp.SerializedIdentity{}
+	err := proto.Unmarshal(id, si)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to unmarshal to msp.SerializedIdentity{}")
+	}
+	block, _ := pem.Decode(si.IdBytes)
+	if block == nil {
+		return "", errors.New("bytes are not PEM encoded")
+	}
+	switch block.Type {
+	case "CERTIFICATE":
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return "", errors.WithMessage(err, "pem bytes are not cert encoded ")
+		}
+		return cert.Subject.CommonName, nil
+	default:
+		return "", errors.Errorf("bad block type %s, expected CERTIFICATE", block.Type)
+	}
+}
+
+func GetRevocationHandle(id []byte) ([]byte, error) {
+	si := &msp.SerializedIdentity{}
+	err := proto.Unmarshal(id, si)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal to msp.SerializedIdentity{}")
+	}
+	block, _ := pem.Decode(si.IdBytes)
+	if block == nil {
+		return nil, errors.New("bytes are not PEM encoded")
+	}
+	switch block.Type {
+	case "CERTIFICATE":
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, errors.WithMessage(err, "pem bytes are not cert encoded ")
+		}
```

Comment:
Up to this point, the method is similar to GetEnrollmentId. We could have a function:
```
func getCertificate(id []byte) (*x509.Certificate, error)
```

Reply:
👍